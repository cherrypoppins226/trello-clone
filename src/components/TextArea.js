import React from "react";
import { findDOMNode } from "react-dom";
import { withStyles } from "material-ui/styles";
import PropTypes from "prop-types";

const styles = {
  root: {
    border: 0,
    resize: "none",
    borderRadius: 2,
    background: "transparent",
    padding: "0.1em 0.2em",
    "&:focus": {
      outlineWidth: 2,
      background: "white",
      boxShadow: "0px 0px 2px 0px"
    }
  }
};

const resize = (node1, node2) =>
  (node1.style.height = `${node2.scrollHeight}px`);

const leaveOnEnter = event => event.keyCode === 13 && event.target.blur();

class TextArea extends React.Component {
  componentDidMount() {
    this.node = findDOMNode(this);
    resize(this.node, this.node);
  }

  render() {
    const { className, classes, value, ...props } = this.props;
    return (
      <textarea
        className={[className, classes.root].join(" ")}
        defaultValue={value}
        onInput={e => resize(this.node, e.target)}
        onKeyDown={leaveOnEnter}
        rows={1}
        {...props}
      />
    );
  }
}

const View = withStyles(styles)(TextArea);

View.propTypes = {
  value: PropTypes.string
};

export default View;
