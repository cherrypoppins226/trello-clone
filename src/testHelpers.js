import { render as renderer } from "react-testing-library";
import { createContext } from "react-cosmos-loader";

export const render = (fixture, proxies = []) => {
  const { mount, ...context } = createContext({ renderer, fixture, proxies });

  return (async () => {
    await mount();
    return { ...context.getWrapper(), ...context, rerender: mount };
  })();
};

export const NativeEvents = {
  mouse: {
    click: new MouseEvent("click", { bubbles: true, cancelable: true })
  }
};

export const getByRole = (node, role) => {
  return node.querySelector(`[role="${role}"]`);
};

export const getByAriaLabelled = (node, id) => {
  return node.querySelector(`[aria-labelledby="${id}"]`);
};

export const getByAriaDescribed = (node, id) => {
  return node.querySelector(`[aria-describedby="${id}"]`);
};
