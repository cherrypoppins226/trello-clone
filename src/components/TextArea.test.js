import React from "react";
import { fireEvent } from "react-testing-library";
import { render, renderIntoDocument } from "../testHelpers.js";
import TextArea from "./Textarea";

it("initial render", () => {
  const { container } = render(<TextArea onClose={jest.fn()} />);
  expect(container).toMatchSnapshot();
});

it("calls onClose when clicked outside", () => {
  const onClose = jest.fn();
  renderIntoDocument(<TextArea onClose={onClose} />);
  const click = new MouseEvent("click", {
    bubbles: true,
    cancelable: true
  });
  fireEvent(document.body, click);
  expect(onClose).toHaveBeenCalledWith(click);
});

it("renders with given value and rows", () => {
  const rows = 5;
  const value = "value";
  const { container } = render(
    <TextArea rows={rows} value={value} onClose={() => {}} />
  );
  const textarea = container.querySelector("textarea");
  expect(textarea.value).toBe(value);
  expect(textarea.rows).toBe(rows);
});
