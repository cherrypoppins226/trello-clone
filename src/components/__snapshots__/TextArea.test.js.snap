// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
<div>
  <textarea
    class=" Textarea-container-1"
    rows="1"
    style="height: 0px;"
  />
</div>
`;
